net-utils/src/ip_echo_client.rs

```diff
@@ -106,39 +107,38 @@ fn parse_response(
 
 pub(crate) const DEFAULT_RETRY_COUNT: usize = 5;
 
-/// Checks if all of the provided TCP/UDP ports are not reachable by the machine at
+/// Checks if all of the provided TCP ports are reachable by the machine at
 /// `ip_echo_server_addr`. Tests must complete within timeout provided.
-/// Tests will run in parallel when possible.
-pub(crate) async fn do_verify_reachable_ports(
+/// Tests will run concurrently to avoid head-of-line blocking. Will return false on any error.
+pub(crate) async fn verify_reachable_tcp_ports(
     ip_echo_server_addr: SocketAddr,
-    tcp_listeners: Vec<(u16, TcpListener)>,
-    udp_sockets: &[&UdpSocket],
+    listeners: Vec<(u16, TcpListener)>,
     timeout: Duration,
-    udp_retry_count: usize,
 ) -> bool {
-    info!(
-        "Checking that tcp ports {:?} are reachable from {:?}",
-        tcp_listeners, ip_echo_server_addr
-    );
+    let mut checkers = Vec::new();
```

Comment:
You can reorganize the code to use `JoinSet`, see https://gist.github.com/KirillLykov/d448ebe7c3ce738a45da47d061c7c2ad

Of course, if it appear to be a better way from your point of view

Reply:
JoinSet can not be used here since we need to have listening_addr available for a task that would not join (so we can unblock a port via connect call).

Reply:
Do you mean `TcpStream::connect_timeout(&listening_addr, timeout).unwrap();`?

Reply:
yes. it is a dirty hack but it has worked so far and I do not know of a better way without breaking API